src/Application.php

```diff
@@ -285,15 +284,17 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
                 if (OutputInterface::VERBOSITY_VERBOSE <= $output->getVerbosity()) {
                     throw $e;
                 }
+
+                return $event->getExitCode();
             } else {
                 throw $event->getException();
             }
-        } finally {
-            $event = new ConsoleTerminateEvent($command, $input, $output, $exitCode);
-            $this->dispatcher->dispatch(ConsoleEvents::TERMINATE, $event);
         }
 
-        return $exitCode;
+        $event = new ConsoleTerminateEvent($command, $input, $output, $exitCode);
+        $this->dispatcher->dispatch(ConsoleEvents::TERMINATE, $event);
+
+        return $event->getExitCode();
```

Comment:
@sstok Sorry I didn't actually go through the code properly, and didn't see the `$exitCode` is set on the `ConsoleTerminateEvent` before the event is dispatched. But after looking at it more closely it makes more sense now :)